tools/proofers/processtext.php

```diff
@@ -306,6 +306,82 @@ function leave_spellcheck_mode( $ppage )
     }
 }
 
+function attempt_to_save_as_done($ppage, $text_data, $and_do_another)
+// This is only an attempt, because a daily page limit might block the save,
+// or prevent further saves.
+// If there's a problem, this function does not return to the caller.
+{
+    global $code_url, $pguser, $projectid, $proj_state;
+
+    $round = get_Round_for_project_state($proj_state);
+
+    list($saved, $dpl_reached) = $ppage->attemptSaveAsDone($text_data, $pguser);
+
+    if (!$dpl_reached)
+    {
+        assert($saved);
+        return; // to let the caller do the appropriate normal thing
+    }
+
+    assert($dpl_reached);
+
+    if ($saved)
+    {
+        $title = _("Saved, but at limit");
+        $sentence = sprintf(
+            _("Your page has been saved as 'Done'. However, you have now reached the daily page limit for %s."),
+            $round->id
+        );
+    }
+    else
+    {
+        $ppage->saveAsInProgress($text_data, $pguser);
+        $title = _("Already at limit");
+        $sentence = sprintf(
+            _("Your page was saved as 'In Progress' rather than 'Done', because you have already reached the daily page limit for %s."),
+            $round->id
+        );
+    }
+
+    slim_header( $title );
+
+    echo "<p>$sentence</p>\n";
+    
+    echo "<p>"
+        . sprintf(
+            _("You will not be allowed to save any more pages in %s until the next server midnight."),
+            $round->id
+          )
+        . "</p>\n";
```

Comment:
This seems like a good place to plug SR, as well as adding a link to the SR page in the listing below.

Reply:
Fair point. Let's leave this without SR.